R/corrpart.b.R

```diff
@@ -246,72 +233,59 @@ corrPartClass <- R6::R6Class(
-
-                    res1 <- try(cor.test(var1, var2, method='pearson'))
-                    res2 <- try(cor.test(var1, var2, method='spearman'))
-                    res3 <- list()
-                    if (self$options$kendall)
-                        res3 <- try(cor.test(var1, var2, method='kendall'))
-
-                } else if (type == "part") {
-
-                    res1 <- try(ppcor::pcor.test(var1, var2, controls, method='pearson'))
-                    res2 <- try(ppcor::pcor.test(var1, var2, controls, method='spearman'))
-                    res3 <- list()
-                    if (self$options$kendall)
-                        res3 <- try(ppcor::pcor.test(var1, var2, controls, method='kendall'))
-
-                } else {
-
-                    res1 <- try(ppcor::spcor.test(var1, var2, controls, method='pearson'))
-                    res2 <- try(ppcor::spcor.test(var1, var2, controls, method='spearman'))
-                    res3 <- list()
-                    if (self$options$kendall)
-                        res3 <- try(ppcor::spcor.test(var1, var2, controls, method='kendall'))
-
-                }
-
-                if ( ! base::inherits(res1, 'try-error')) {
-                    results$r  <- res1$estimate
-                    results$rp <- res1$p.value
-                } else {
-                    results$r <- NaN
-                    results$rp <- NaN
-                }
-
-                if ( ! base::inherits(res2, 'try-error')) {
-                    results$rho <- res2$estimate
-                    results$rhop <- res2$p.value
-                } else {
-                    results$rho <- NaN
-                    results$rhop <- NaN
-                }
-
-                if ( ! base::inherits(res3, 'try-error')) {
-                    results$tau <- res3$estimate
-                    results$taup <- res3$p.value
-                } else {
-                    results$tau <- NaN
-                    results$taup <- NaN
-                }
-
-                results$n <- length(data[[1]])
+                for (method in c('pearson', 'spearman', 'kendall')) {
+                    # calculate 'kendall' only if the respective tick box is set
+                    if (method == 'kendall' && self$options$kendall == F) { next }
+                        
+                    try({
+                        # the following code took some inspiration from the ppcor-package (https://cran.r-project.org/web/packages/ppcor)
+                        # calculate covariance matrix and invert it
+                        cvx <- cov(rdta, method = method)
+	                    if (det(cvx) < .Machine$double.eps) { icvx <- ginv(cvx) } else { icvx <- solve(cvx) }
+	                
+	                    # calculation of the partial correlation coefficient 
+                        if (type == 'part') {
+                            results[ifelse(method == 'pearson', 'r', ifelse(method == 'spearman', 'rho', 'tau')), 1] <- -cov2cor(icvx)[1, 2]
```

Comment:
I'm usually not a big fan of nested ifelse statements (because it becomes difficult to read), so if there's a different way to do this, I'd prefer it. For example you could assign the name of the statistic (`r`, `rho`, or `tau`) before the if statement so you can use it here, and the semi-partial correlation. 

So, for example:
```
if (method == 'pearson')
    statName <- 'r'
else if (method == 'spearman')
    statName <- 'rho'
else
    statName <- 'tau'
```

And then:
```
results[statName, 1] <- -cov2cor(icvx)[1, 2]
```

Reply:
Yes, much easier indeed...